### PR TITLE
Add support for the new username system

### PR DESCRIPTION
**Pull request by starshine-sys**
Adds what is currently documented in https://github.com/discord/discord-api-docs/pull/6130. The documentation isn't merged yet, but this has already started rolling out to normal users.

### CODE REVIEW

**Comment by diamondburned** [comment]
Is the `GlobalName` here the display name? AFAIK the new global username is kept the same in the `Username` field?

**Comment by starshine-sys** [comment]
Yeah, `GlobalName` is the display name--I'm pretty sure the field was originally called `display_name` but it got changed before the rollout.

**Comment by diamondburned** [comment]
> I'm pretty sure the field was originally called `display_name` but it got changed before the rollout.

Is there a good reason for this? Should we call it `DisplayName` on our end (to match the client) or keep? Does Discord intend to eventually remove display names?

**Comment by starshine-sys** [comment]
I can't think of any good reason for it being changed other than maybe reducing confusion with server nicknames, so `DisplayName` is probably a better name for it, yeah. I doubt Discord will remove display names because it's prominently shown in the UI and mentioned in all their support articles.

**Comment by diamondburned** [comment]
Can you make the change? I'll approve the PR afterwards.

File: discord/user.go [hunk 1]
```diff
@@ -8,8 +8,9 @@ import (
 type User struct {
 	ID            UserID `json:"id"`
 	Username      string `json:"username"`
-	Discriminator string `json:"discriminator"`
+	Discriminator string `json:"discriminator"` // This is "0" if the user has migrated to the new username system.
 	Avatar        Hash   `json:"avatar"`
+	GlobalName    string `json:"global_name"`
```
Comment (diamondburned) [inline discord/user.go]: `s/GlobalName/DisplayName/`

**Review by diamondburned — CHANGES_REQUESTED** [review]

**Review by diamondburned — APPROVED** [review]

**Comment by diamondburned** [comment]
In hindsight, `GlobalName` would've been better, since that would allow a `DisplayName() string` method. I will add a `DisplayOrUsername() string` though.

**Pull request merged**